overlord/ifacestate/handlers.go

```diff
@@ -1259,6 +1259,83 @@ func (m *InterfaceManager) doGadgetConnect(task *state.Task, _ *tomb.Tomb) error
 	return nil
 }
 
+// doHotplugConnect creates task(s) to (re)create connections in response to hotplug "add" event.
+func (m *InterfaceManager) doHotplugConnect(task *state.Task, _ *tomb.Tomb) error {
+	st := task.State()
+	st.Lock()
+	defer st.Unlock()
+
+	conns, err := getConns(st)
+	if err != nil {
+		return err
+	}
+
+	ifaceName, hotplugKey, err := getHotplugAttrs(task)
+	if err != nil {
+		return fmt.Errorf("internal error: cannot get hotplug task attributes: %s", err)
+	}
+
+	// find old connections for slots of this device - note we can't ask the repository since we need
+	// to recreate old connections that are only remembered in the state.
+	connsForDevice := findConnsForHotplugKey(conns, ifaceName, hotplugKey)
+
+	// we see this device for the first time (or it didn't have any connected slot before)
+	if len(connsForDevice) == 0 {
+		return m.autoconnectNewDevice(task, ifaceName, hotplugKey)
```

Comment:
Shouldn't we always consider if there are possible autoconnection that can happen when a device is (re)connected? after having reestablished already present connections?

Consider installing a snap  that would autoconnect while a device is momentarely disconnected,  vs doing when is connected. Should they behave differently after the device is reconnected?

Reply:
I think you're right about the first part. There might be new auto-connection candidates when we see same device again. I didn't consider this, thank you.

I'm unclear about the second paragraph, I think we need to discuss that part, I'm not sure how/if this is related to the first concern.

Reply:
Updated, with a few extra unit tests.